feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/sdk/discovery/ble/BleLedgerDeviceDiscoveryService.kt

```diff
@@ -67,5 +71,21 @@ class BleLedgerDeviceDiscoveryService(
         override fun onScanFailed(errorCode: Int) {
             errors.tryEmit(BleScanFailed(errorCode))
         }
+
+        private fun ScanResult.getLedgerDeviceType(): LedgerDeviceType? {
+            val searchingServiceUUIDs = this.scanRecord?.serviceUuids
+                .orEmpty()
+                .map { it.uuid }
+
+            for (ledgerDeviceType in LedgerDeviceType.values()) {
+                val bleInfo = ledgerDeviceType.bleDevice as? BleDevice.Supported ?: continue
+
+                if (bleInfo.serviceUuid in searchingServiceUUIDs) {
+                    return ledgerDeviceType
+                }
+            }
```

Comment:
We can move this to `Companion` of `LedgerDeviceType` so we can do `LedgerDeviceType.findByServiceUUID(uuid)`